docs/server/source/production-deployment-template-tendermint/architecture.rst

```diff
@@ -0,0 +1,204 @@
+  +--------------------------------------------------------------------------------------------------------------------------------------+
+                  |                                                                                     |
+                  |                                                                                     |
+                  v                                                                                     v
+
+  +------------------------------------+                                                +------------------------------------+
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  |      3Scale                        |                                                |    MongoDB Cloud                   |
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  +------------------------------------+                                                +------------------------------------+
+
+
+
+.. note::
+  The arrows in the diagram represent the client-server communication. For
+  example, A-->B implies that A initiates the connection to B.
+  It does not represent the flow of data; the communication channel is always
+  fully duplex.
+
+
+NGINX: Entrypoint and Gateway
+-----------------------------
+
+We use an NGINX as HTTP proxy on port 443 (configurable) at the cloud
+entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. HTTPS Termination: The HTTPS connection does not carry through all the way
+   to BigchainDB and terminates at NGINX for now.
+
+#. Request Routing: For HTTPS connections on port 443 (or the configured BigchainDB public api port),
+   the connection is proxied to:
+
+   #. OpenResty Service if it is a POST request.
+   #. BigchainDB Service if it is a GET request.
+
+
+We use an NGINX TCP proxy on port 27017 (configurable) at the cloud
+entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. Request Routing: For connections on port 27017 (or the configured MongoDB
+   public api port), the connection is proxied to the MongoDB Service.
```

Comment:
Who talks to MongoDB Cloud, and on what port?

Reply:
Agents talk to the MongoDB cloud on port `443` and for agents to talk to the MongoDB instance we need the port `27017` .

Reply:
Please clarify that in the text of this page. It's not clear to me why there's a line connecting "Rate Limiting Logic" (from the outside world) to "MongoDB Service". Why would anyone external need to communicate to the MongoDB Service? (Assuming no MongoDB replica set.)

On the diagram, you could also add a "443" label beside the line connecting MongoDB Cloud to "MongoDB Monitoring Agent" (if I understand that correctly). (One might reasonably wonder what port is used for that.)